### PR TITLE
feat(ui-ext): make 'mousehide' into proper ui_option

### LINKED ISSUE #14955: make 'mousehide' into proper ui_option
- `nvim --version`: NVIM v0.5.0-dev+1455-gf83c25942
- Operating system/version: Windows 10 20H2 19042.1052
- Terminal name/version: GUI
- `$TERM`: nvim

[x] `vim -u DEFAULTS` (version: 8.2) behaves differently

### Steps to reproduce using `nvim -u NORC`
```
nvim -u NORC
echo nvim_get_option("mousehide")
```

### Actual behaviour
On vim 8.2 mousehide option can be used in OptionSet autocmd but in neovim it doesn't work.
And nvim_get_option_info works with mousehide but nvim_get_option doesn't work.

### Expected behaviour
Get the value of the mousehide.

### PR DESCRIPTION
**Pull request by jaehoonhwang**
Fixes #14955

Tested using 

```sh
nvim -u NORC
echo nvim_get_option("mousehide")
```

Tests; Passed functional test; `TEST_FILE=test/functional/ui/options_spec.lua make functionaltest`

This PR just adds an option to be visible, there is no actual implementation as per #14955.

### CODE REVIEW

**Comment by jaehoonhwang** [comment]
Oops fixing the tests

**Comment by zeertzjq** [comment]
Needs `redraw = { 'ui_option' }`

**Comment by jaehoonhwang** [comment]
Done! 
PTAL,

File: src/nvim/options.lua [hunk 1]
```diff
@@ -5452,9 +5452,11 @@ return {
       ]=],
       enable_if = false,
       full_name = 'mousehide',
+      redraw = { 'ui_option' },
       scope = { 'global' },
       short_desc = N_('hide mouse pointer while typing'),
       type = 'bool',
+      varname = 'p_mouseh',
```
Comment (zeertzjq) [inline src/nvim/options.lua:5459]: The abbreviation is `mh`, so `p_mh` is better for consistency.

File: src/nvim/option_vars.h [hunk 2]
```diff
@@ -571,6 +571,7 @@ EXTERN char *p_mouse;           ///< 'mouse'
 EXTERN char *p_mousem;          ///< 'mousemodel'
 EXTERN int p_mousemev;          ///< 'mousemoveevent'
 EXTERN int p_mousef;            ///< 'mousefocus'
+EXTERN int p_mouseh;            ///< 'mousehide'
```
Comment (zeertzjq) [inline src/nvim/option_vars.h:574]:
```suggestion
EXTERN int p_mh;                ///< 'mousehide'
```

File: src/nvim/options.lua (hunk 1 above)
Comment (zeertzjq) [inline src/nvim/options.lua:5459]:
```suggestion
      varname = 'p_mh',
```

**Comment by zeertzjq** [comment]
Strangely this is hanging on `--embed UI can pass stdin` test on Windows and I don't know why it happens.

**Comment by zeertzjq** [comment]
And it's hanging at the same test as #24888

**Review by justinmk — APPROVED** [review]

**Pull request merged**